Expensify/App
# IOU Invalid Phone number error handled

### Details


### Fixed Issues

$ https://github.com/Expensify/App/issues/7688

### Tests
1. Click on FAB ➕  Button -> Split Bill
2. Enter Amount and press Next
3. Search and select a three-digit phone number
4. Click Next and press the Split Bill button
5. You should see the following error message for the phone number
`Please enter a valid phone number, with the country code (e.g. +1234567890)`

#### Success case:
6. Now press next.
7.  Deselect the invalid number.
8. Search and select a valid phone number (Ex: +1234567890)
9. Click Next and press the Split Bill button
10. Split should happen successfully and open to the chat!


### PR Review Checklist

#### Contributor (PR Author) Checklist
- [x] I ran the tests & verify they passed on:

#### PR Reviewer Checklist

### QA Steps
1. Click on FAB ➕  Button -> Split Bill
2. Enter Amount and press Next
3. Search and select a three-digit phone number
4. Click Next and press the Split Bill button
5. You should see the following error message for the phone number
`Please enter a valid phone number, with the country code (e.g. +1234567890)`


#### Success case:
6. Now press next.
7.  Deselect the invalid number.
8. Search and select a valid phone number (Ex: +1234567890)
9. Click Next and press the Split Bill button
10. Split should happen successfully and open to the chat!


### Screenshots


#### Web
https://user-images.githubusercontent.com/85645967/156887210-4ce9caa2-2455-4a90-9cc7-c3e0ca5b6ece.mov

#### Mobile Web
https://user-images.githubusercontent.com/85645967/156887282-05413ec8-210e-44de-a676-5cc08ac58ed0.mp4

#### Desktop
https://user-images.githubusercontent.com/85645967/156887303-89effabf-bd33-4a96-819f-3611cf28c8a4.mov

#### iOS
https://user-images.githubusercontent.com/85645967/156893906-8edd3b23-31b4-4885-96bd-37b507ef081f.mp4

#### Android
https://user-images.githubusercontent.com/85645967/156887336-1ce29f0d-27c8-43a4-9add-7eb5158e9fb6.mov
